### PR TITLE
MGMT-16332: fix issue validating pre-release versions

### PR DESCRIPTION
**Pull request by eifrach**
- Fixing compering pre release version were failing
- Adding test cases for pre release versions

## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [X] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/operators/lvm/lvm_operator_test.go [hunk 1]
```diff
@@ -108,6 +108,14 @@ var _ = Describe("Lvm Operator", func() {
 				&common.Cluster{Cluster: models.Cluster{HighAvailabilityMode: &fullHaMode, Hosts: []*models.Host{hostWithSufficientResources, hostWithSufficientResources}, OpenshiftVersion: operator.config.LvmMinMultiNodeSupportVersion}},
 				api.ValidationResult{Status: api.Success, ValidationId: operator.GetHostValidationID()},
 			),
+			table.Entry("High Availability Mode Full with pre-release version",
+				&common.Cluster{Cluster: models.Cluster{HighAvailabilityMode: &fullHaMode, Hosts: []*models.Host{hostWithSufficientResources, hostWithSufficientResources}, OpenshiftVersion: "4.15.0-rc0"}},
+				api.ValidationResult{Status: api.Success, ValidationId: operator.GetHostValidationID()},
+			),
+			table.Entry("High Availability Mode Full with higher than LvmMinMultiNodeSupportVersion",
```
Comment (rccrdpccl) [inline internal/operators/lvm/lvm_operator_test.go:115]: Can we add SNO with openshift version > LvmMinMultiNodeSupportVersion?

File: internal/operators/lvm/lvm_operator.go [hunk 2]
```diff
@@ -94,7 +94,7 @@ func (o *operator) ValidateCluster(_ context.Context, cluster *common.Cluster) (
 		if err != nil {
 			return api.ValidationResult{Status: api.Failure, ValidationId: o.GetHostValidationID(), Reasons: []string{err.Error()}}, nil
 		}
-		if ocpVersion.LessThan(minOpenshiftVersionForMultiNodeSupport) && !common.IsSingleNodeCluster(cluster) {
+		if ok, _ := common.BaseVersionLessThan(minOpenshiftVersionForMultiNodeSupport.String(), ocpVersion.String()); ok {
```
Comment (rccrdpccl) [inline internal/operators/lvm/lvm_operator.go:97]: Check about being SNO disappeared, is that intended?

IIUC in this case SNO with version greater than minimum for multi node would pass

File: internal/operators/lvm/lvm_operator_test.go (hunk 1 above)
Comment (eifrach) [reply internal/operators/lvm/lvm_operator_test.go:115]: already there - the naming is a big confusing   
```go
noneHaMode := models.ClusterHighAvailabilityModeNone
```
https://github.com/openshift/assisted-service/blob/a2095745415dd0d8b680808b74924eb3bccd0b47/internal/operators/lvm/lvm_operator_test.go#L123-L130

Comment (rccrdpccl) [reply internal/operators/lvm/lvm_operator_test.go:115]: yep sorry I've missed it :sweat:

**Pull request merged**